runtime/slice.go

```diff
@@ -156,6 +156,58 @@ func initSliceType(map[string]*Object) {
 	SliceType.slots.Repr = &unaryOpSlot{sliceRepr}
 }
 
+func sliceIndex(f *Frame, o *Object) (i int64, raised *BaseException) {
```

Comment:
Ah, sorry, I copy pasted that code in golang.org and tried that, got those results.
https://play.golang.org/p/KLg6Io_tO5
grumpy implementation is okay